### ISSUE TITLE
Is it possible to serialize continuations?

### ISSUE THREAD

**Issue by oravecz** [vsilaev/tascalate-javaflow#1]
Like to serialize/deserialize continuations. Possible?

**Comment by vsilaev** [comment]
Technically -- yes, it's possible. All related tascalate-javaflow classes are serializable. Like in the original JavaFlow library, btw.

Practically it's hard. You should have only serializable objects in the call chain, all serializable parameters of all methods involved and all serializable variables (visible at the point your code invokes Continuation.suspend). This requirement is pretty hard to follow -- there are a lot of classes in JDK these are not serializable, but used intensively, like Java 8 Streams or java.util.Iterator implementations. I even did some experiments how to wrap out-of-the-box Iterator-s into serializable form -- please see https://github.com/vsilaev/serializable-iterators. Meanwhile, alternative serialization libraries, like Kryo, support serialization of such classes.
